airbyte/_processors/sql/base.py

```diff
@@ -548,6 +548,8 @@ def write_stream_data(
             finally:
                 self._drop_temp_table(temp_table_name, if_exists=True)
 
+        progress.log_stream_finalized(stream_name)
```

Comment:
This was the "fix" part of the PR. The rest of the changes are mostly improvements to testing.